Sources/PointFree/Logger.swift

```diff
@@ -1,43 +1,99 @@
+import Foundation
 #if os(Linux)
 import Glibc
 #else
 import Darwin.C
 #endif
 
-public struct Logger {
-  private(set) var level: Level = .debug
-  private(set) var logger: (String) -> () = { print($0) }
+public class StandardFileHandle: TextOutputStream {
+  fileprivate let handle: FileHandle
+
+  public static let error = StandardFileHandle(handle: .standardError)
+  public static let output = StandardFileHandle(handle: .standardOutput)
+  public static let null = StandardFileHandle(handle: .nullDevice)
+
+  public init(handle: FileHandle) {
+    self.handle = handle
+  }
+
+  public func write(_ string: String) {
+    self.handle.write(Data(string.utf8))
+  }
+}
+
+public class Logger {
+  private let level: Level
+  private var output: StandardFileHandle
+  private var error: StandardFileHandle
+
+  init(
+    level: Level = .debug,
+    output: StandardFileHandle = .output,
+    error: StandardFileHandle = .error) {
 
-  init(level: Level = .debug, logger: @escaping (String) -> () = { print($0) }) {
     self.level = level
-    self.logger = logger
+    self.output = output
+    self.error = error
   }
 
-  public func log<A>(_ level: Level, _ message: @autoclosure () -> A) {
+  public func log<A>(
+    _ level: Level,
+    _ message: @autoclosure () -> A,
+    file: StaticString = #file,
+    line: UInt = #line) {
+
+    let file = String(String(describing: file).split(separator: "/").last!)
```

Comment:
doing this so that logs look nice and short:

```
[debug] SiteMiddleware.swift:11: [Request] GET http://localhost:8080/
[debug] SiteMiddleware.swift:11: [Time]    88ms
```

Turns out SPM can't even handle multiple files with the same name, so I think this will sufficiently disambiguate, at least for now.